src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -268,15 +269,19 @@ private void filterSensitiveFeedbackInCourseExercise(Participation participation
         results.forEach(result -> {
             boolean isBeforeDueDateOrAutomaticAndBeforeLatestDueDate = participationBeforeDueDate
                     || (AssessmentType.AUTOMATIC.equals(result.getAssessmentType()) && beforeLatestDueDate);
-            result.filterSensitiveFeedbacks(isBeforeDueDateOrAutomaticAndBeforeLatestDueDate);
+            if (Hibernate.isInitialized(result.getFeedbacks())) {
+                result.filterSensitiveFeedbacks(isBeforeDueDateOrAutomaticAndBeforeLatestDueDate);
+            }
```

Comment:
why did you add this check here? if it is possible that hibernate is not initialized, shouldn't there be an else clause because otherwise the sensitive feedback does not get filtered?

Reply:
I talked to @Strohgelaender and this check is if the results have no feedback. For example for the query in ProgrammingExerciseStudentParticipationRepository.java I'm not fetching the feedback. If the check isn't there there is a Error